scipy/stats/_continuous_distns.py

```diff
@@ -6942,6 +6942,116 @@ def _stats(self, a):
     def _entropy(self, a):
         return 1 - 1.0/a - np.log(a)
 
+    def fit(self, data, *args, **kwds):
+        data, fshape, floc, fscale = _check_fit_input_parameters(self, data,
+                                                                 args, kwds)
+        # ensure that any fixed parameters don't violate constraints of the
+        # distribution before continuing. The support of the distribution
+        # is `0 < (x - loc)/scale < 1`.
+        if floc is not None:
+            if not data.min() >= floc:
+                raise FitDataError('powerlaw', 0, 1)
+            if fscale is not None and not data.max() <= floc + fscale:
+                raise FitDataError('powerlaw', 0, 1)
+
+        if fscale is not None and fscale <= 0:
+            raise Exception("Negative or zero `fscale` is outside the range "
+                            "allowed by the distribution.")
+
+        def shape_universal(data, loc, scale):
+            # The first-order necessary condition on `shape` can be solved in
+            # closed form
+            mask = data != loc
+            return -len(data) / np.sum(np.log((data[mask] - loc)/scale))
+
+        def scale_universal(data, loc):
+            # analytical solution for `scale` based on the location.
+            return data.max() - loc
+
+        # First, attempt to fit under the assumption that `shape <= 1`.
+        def universal_slte1(data):
+            loc = floc if floc is not None else np.nextafter(data.min(),
+                                                             -np.inf)
+            scale = fscale or np.nextafter(scale_universal(data, loc),
+                                           np.inf)
+            shape = fshape or shape_universal(data, loc, scale)
+            return shape, loc, scale
+
+        shape, loc, scale = universal_slte1(data)
+
+        # If the `shape <= 1`, then fitting has succeeded with the assumption
+        # `shape <= 1`.
+        if 0 < shape <= 1:
+            return shape, loc, scale
+
+        # if `shape > 1`, then we need to fit under the assumption that
+        # `shape > 1` and redefine analytical functions for this new
+        # assumption.
+
+        def location_sgt1(data, scale):
+            # analytical solution for `loc` when `shape > 1` based on the
+            # scale.
+            return data.max() - scale
+
+        # if the location or scale is fixed, an analytical solution is
+        # available.
+        if {floc, fscale} != {None}:
+            if floc is not None:
+                scale = fscale or scale_universal(data, floc)
+                return ((fshape or shape_universal(data, floc, scale)), floc,
+                        scale)
```

Comment:
For consistency (style similar to line above) and readability:
```suggestion
                shape = fshape or shape_universal(data, floc, scale)
                return (shape, floc, scale)
```
Same idea below.